lib/recurly/BaseClient.js

```diff
@@ -47,7 +48,16 @@ class BaseClient {
     return path
   }
 
+  _validateOptions (options) {
+    Object.keys(options).forEach(key => {
+      if (!ALLOWED_OPTIONS.includes(key)) {
+        throw new Error(`'${key}' is not a valid option`)
```

Comment:
Could we include the contents of the `ALLOWED_OPTIONS` array in this message?

Reply:
Also, is there a base error type we can use so this can be identified as a Recurly error?

Reply:
Yeah, including the allowed options is a good idea. The `ApiError` is the top most Recurly error and the naming of it didn't feel 100% right to me, so I defaulted to `Error`. I've updated